File: Source/Aladdin.ASN1.Java/src/aladdin/asn1/Boolean.java
package aladdin.asn1;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Булевое значение
///////////////////////////////////////////////////////////////////////////
public final class Boolean extends AsnObject
{
    private static final long serialVersionUID = 5029530852418640069L;
    
    // проверить допустимость типа
    public static boolean isValidTag(Tag tag) { return tag.equals(Tag.BOOLEAN); }
    
    // булевы значения 
    public static Boolean TRUE  = new Boolean(true );
    public static Boolean FALSE = new Boolean(false);

    // конструктор при раскодировании
    public Boolean(IEncodable encodable) throws IOException
    {
        // инициализировать объект
    	super(encodable); init(); 
    }
    // сериализация
    @Override protected void readObject(ObjectInputStream ois) throws IOException 
    {
        // прочитать объект
        super.readObject(ois); init(); 
    }    
    // инициализировать объект
    private void init() throws IOException
    {
        // проверить корректность способа кодирования
		if (pc() != PC.PRIMITIVE) throw new IOException();

		// проверить корректность объекта
		if (content().length != 1) throw new IOException();

		// сохранить булевое значение
		this.value = (content()[0] != 0);
    }
    // конструктор при закодировании
    public Boolean(boolean value) { super(Tag.BOOLEAN); this.value = value; }
    
    // способ кодирования для DER-кодировки
    @Override protected final PC derPC() { return PC.PRIMITIVE; }

    // содержимое объекта
    @Override protected final byte[] derContent()
    {
		// вернуть содержимое объекта
		return new byte[] { value ? (byte)0xFF : (byte)0x00 };
    }
    // булевое значение
    public final boolean value() { return value; } private boolean value;
}

File: Source/Aladdin.ASN1/ASN1/Boolean.cs
using System;
using System.IO;
using System.Runtime.Serialization;

namespace Aladdin.ASN1
{
	///////////////////////////////////////////////////////////////////////////
	// Булевое значение
	///////////////////////////////////////////////////////////////////////////
	[Serializable]
	public sealed class Boolean : AsnObject
	{
        // проверить допустимость типа
        public static bool IsValidTag(Tag tag) { return tag == Tag.Boolean; }
    
		// булевы значения 
		public static readonly Boolean True  = new Boolean(true );
		public static readonly Boolean False = new Boolean(false); 

		// конструктор при сериализации
        private Boolean(SerializationInfo info, StreamingContext context) 
		
			// инициализировать объект
			: base(info, context) { Init(); } private void Init()
        {
			// проверить корректность способа кодирования
			if (PC != PC.Primitive) throw new InvalidDataException();

			// проверить корректность объекта
			if (Content.Length != 1) throw new InvalidDataException();

			// сохранить булевое значение
			this.value = (Content[0] != 0); 
        }
		// конструктор при раскодировании
		public Boolean(IEncodable encodable) : base(encodable) { Init(); }

		// конструктор при закодировании
		public Boolean(bool value) : base(Tag.Boolean) { this.value = value; }

		// способ кодирования для DER-кодировки
		protected override PC DerPC { get { return PC.Primitive; } }

		// содержимое объекта
		protected override byte[] DerContent { get 
		{
			// вернуть содержимое объекта
			return new byte[] { value ? (byte)0xFF : (byte)0x00 } ; 
		}}
 		// булевое значение
		public bool Value { get { return value; } } [NonSerialized] private bool value;
	}
}
